server/src/main/java/org/elasticsearch/action/support/ThreadedActionListener.java

```diff
@@ -19,21 +19,22 @@
 /**
  * An action listener that wraps another action listener and dispatches its completion to an executor.
  */
-public final class ThreadedActionListener<Response> extends ActionListener.Delegating<Response, Response> {
+public final class ThreadedActionListener<Response> implements ActionListener<Response> {
```

Comment:
Just curious: why does this need to change? Isn't the implementation still delegating?

Reply:
We're not using anything from `DelegatingActionListener` here - we completely override both `onFailure` and `toString`.